runner.py

```diff
@@ -0,0 +1,200 @@
+import asyncio
+from collections import namedtuple
+import logging
+import functools
+from typing import Any, Optional
+import urllib.parse
+
+import docker
+import aiohttp as ah
+from database import Database
+
+from config import settings
+
+doc = docker.from_env()
+
+logger = logging.getLogger(__name__)
+
+
+async def run_cmd(cmd: str, env: dict[str, str], vols: dict[str, dict[str, str]]) -> str:
+    """
+    Thin wrapper to simplify the Docker interface & provide secure defaults.
+    """
+    loop = asyncio.get_event_loop()
+    raw_out = await loop.run_in_executor(
+        None,
+        functools.partial(
+            doc.containers.run,
+            settings.docker.container_ref,
+            cmd,
+            environment=env,
+            remove=True,
+            stdout=True,
+            stderr=True,
+            mem_limit="8g",
+            network_mode="none",
+            volumes=vols,
+        ),
+    )
+    out: str = raw_out.decode("utf-8")
+    return out
+
+
+async def bg_update() -> None:
+    """
+    Background update task. Run periodically, or metadata about tags in the DB won't be correct.
+    Definitely run on startup; startup is when the bench format change will first be noticed.
+    """
+
+    image = settings.docker.container_ref
+
+    async with ah.ClientSession() as session:
+        # TODO: get_remote_tags is supposed to be paginated...
+        tags = await get_remote_tags(session, image)
+        new_versions: set[str] = set()
+        with Database() as db:
+            new_versions = db.pick_new_container_versions(tags)
+        for ver in new_versions:
+            rust_ver = await get_rust_version(image, ver)
+            bench_format, _stamp = ver.split(".")
+            with Database() as db:
+                # TODO: Figure out something to do with the bench_dir
+                db.insert_container_version(rust_ver.ver, ver, int(bench_format), b"")
+
+
+# --- Utils ---
+
+RustVersion = namedtuple("RustVersion", "ver hash date")
```

Comment:
could we use typing.NamedTuple here?
```suggestion
from typing import NamedTuple

class RustVersion(NamedTuple):
    ver: str
    hash: str
    date: str
```